src/server/api/routers/webhook.ts

```diff
@@ -143,21 +146,18 @@ export const webhookRouter = createTRPCRouter({
         );
 
         if (response.status === 204) {
-          console.log("Webhook deleted successfully");
           return input.hookID;
         } else if (response.status === 404) {
-          console.log("WebHook doesn't exist", response.status);
-          return false;
+          return 404;
```

Comment:
in this function you're returning either the hook ID or a HTTP status or a boolean? 

while this is possible since JS is dynamically typed, it's generally bad practice outside of some exceptions like `int | float` or `<something> | undefined | null`. 

util/library functions need to be easy to understand and easy to use for other developers - having a return signature be `int | bool` jeopardizes that. other people won't know what the int means and what the bool means without reading the source code of your library. additionally, since the int can mean two different things here (hook ID and status), users have to remember to check the value with no indication from the type system.

one way of doing this is just being explicit with named return types: https://www.typescriptlang.org/play?ssl=26&ssc=2&pln=1&pc=1#code/MYewdgzgLgBAtgTwGYFcwwLwwBQA8BcMADAD4CMJATAJSYB8MA3gFAxswCWSOumGWRWi3YiYAJwCmUFGPTDRCgBYgQAawCSAEUJkiRADSsFMAL5G2JmBIA2ECZ254+WMkPOjJ02U3fHOEAH0Acw4oRRQAIwCJMTEQMUIoMRQJQz8RaABDaQhCAFY9NOMzEUsbOx8-Txk5X2MOQJCwyOjY+MIkTNtUuoUsnMIAFkLe03czM2ZQSFhJCBRrWCxEVDBsQQBuZi4cOYWoADplNS0YAEIsMAXrNxFpiBBrCQPrECDsACJ54GAJCAgPvpxH99kcVBpNNQtmVug5diDFgcGsFQuEojE4mJbux7o9nq93h8mmirG0xIDgfNEf0UBAocwYRV5GxcU8Xm9PmhVGAQAB3dAY+IUvbUqDZWnUBlAA

this is just one way to do it, you can also make it a discriminated union with explicit `kind: 'Success'` or `kind: 'Failure'` etc fields

Reply:
Yeah this is not a very good approach because the user will not really understand what is happening unless they actually look at the source code. The approach you've sent is definitely an improvement but I have another approach that I think will be more intuitive. 

When using [React Query](https://tanstack.com/query/v3/), a powerful asynchronous state management, you receive an object when using queries. The object will look something along the lines of:

```
{
  error: false,
  data: Object,
  status: 202,
  etc...
}
```

I really like this approach because you know exactly what happened and if you get an error, then you will know that your data is null and have the status handy. I feel like this approach is much more understandable and in my current situation, can just return an object with a predefined structure.

Reply:
This is what I've written right now (error handling in the `catch` block). Any thoughts?

```ts
interface WebHookReturnObject {
  error: boolean;
  status: number;
  data: string | null;
}
```

```ts
if (isGitHubAPIError(error)) {
          // if 422: hook already exists in the repository
          if (error.status === 422) {
            return {
              error: true,
              status: 422,
              data: null
            }
          } else if (error.status === 404) {
            return {
              error: true,
              status: 404,
              data: null
            }
          } else {
            return {
              // a possible error can be 401: Bad Credentials if DB accessToken is stale AND user's session is remembered.
              // We require the user to log out and log in.
              error: true,
              status: 401,
              data: null
            }
          }
        }
        return {
          error: true,
          status: 500,
          data: null
        }
      }
      // returning hookID so automation model can grab it. occurs in src/components/FormType.tsx
      return {
        error: false,
        status: 200,
        data: hookID
      }

```

Reply:
So hook ID is actually a string?

Seems reasonable, IDC that much about the exact structure of the response as long as you're not returning a union of primitive types. If I were to really nitpick I would say that data doesn't need to be present in the branches where it's not used, instead of having it always be present and sometimes be null, you can have it maybe be present and if it's present always be non-null. 

```
interface WebHookReturnObject {
  error: boolean;
  status: number;
  data?: string;
}
```
This is a very minor detail, but I think it lets you save a bit of memory by allocating a smaller object when you don't need the extra field. Usage should be pretty similar.

Reply:
I like this nitpick since it does save a small amount of memory and it is an easy change. Will implement this now and merge the PR